### PR TITLE
feat(install): ajax based install system

### LINKED ISSUE #86: Make better install/upgrade view
During my testing of a new 4.3 install, it took a while for the DB to be populated with the required tables and entries.
This can run into timeouts depending on the PHP environment.
So I think we should look into splitting the DB install/upgrade into several smaller subsets, and also provide a status update on where we are at the moment (progress-bar?).

### PR DESCRIPTION
**Pull request by derchrisuk**
closes #86
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/webSPELL/webSPELL/pull/209%23issuecomment-85625226%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/209%23issuecomment-92694052%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/209%23discussion_r28308835%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/209%23discussion_r28310079%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/209%23discussion_r28345160%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/209%23issuecomment-85625226%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22We%20need%20a%20better%20naming%20schema%20for%20the%20called%20functions%22%2C%20%22created_at%22%3A%20%222015-03-24T18%3A05%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Phhere%22%7D%7D%2C%20%7B%22body%22%3A%20%22%60_mysql.php%60%20has%20to%20be%20reverted%22%2C%20%22created_at%22%3A%20%222015-04-14T08%3A32%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2080b76611cde267171cb287e45dc4c986114d69f6%20install/index.php%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/209%23discussion_r28308835%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20is%20just%20%60%3C%21DOCTYPE%20html%3E%60%20in%20HTML5%22%2C%20%22created_at%22%3A%20%222015-04-14T08%3A33%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%2C%20%7B%22body%22%3A%20%22As%20we%20have%20not%20changed%20much%20in%20the%20installer%20it%20should%20be%20like%20it%20is%20now%22%2C%20%22created_at%22%3A%20%222015-04-14T16%3A44%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Phhere%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20install/index.php%3AL60-104%22%7D%2C%20%22Pull%2080b76611cde267171cb287e45dc4c986114d69f6%20install/install.js%2040%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/209%23discussion_r28310079%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22The%20font-Tag%20is%20deprecated%5Cr%5Cn%5Cr%5Cnhttp%3A//www.w3.org/TR/html5/obsolete.html%23non-conforming-features%5Cr%5Cn%5Cr%5Cnput%20that%20stuff%20into%20the%20html%2C%20add%20classes%20and%20then%20use%20%60.show%28%29%60%20instead.%20Why%20don%27t%20you%20use%20bootstrap%20here%3F%20http%3A//getbootstrap.com/components/%23alerts%22%2C%20%22created_at%22%3A%20%222015-04-14T08%3A52%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20install/install.js%3AL1-50%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/209#issuecomment-85625226'>General Comment</a></b>
- <a href='https://github.com/Phhere'></a> We need a better naming schema for the called functions
- <a href='https://github.com/Pascalmh'></a> `_mysql.php` has to be reverted
- [ ] <a href='#crh-comment-Pull 80b76611cde267171cb287e45dc4c986114d69f6 install/index.php 5'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/209#discussion_r28308835'>File: install/index.php:L60-104</a></b>
- <a href='https://github.com/Pascalmh'></a> This is just `<!DOCTYPE html>` in HTML5
- <a href='https://github.com/Phhere'></a> As we have not changed much in the installer it should be like it is now
- [ ] <a href='#crh-comment-Pull 80b76611cde267171cb287e45dc4c986114d69f6 install/install.js 40'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/209#discussion_r28310079'>File: install/install.js:L1-50</a></b>
- <a href='https://github.com/Pascalmh'></a> The font-Tag is deprecated
http://www.w3.org/TR/html5/obsolete.html#non-conforming-features
put that stuff into the html, add classes and then use `.show()` instead. Why don't you use bootstrap here? http://getbootstrap.com/components/#alerts

<a href='https://www.codereviewhub.com/webSPELL/webSPELL/pull/209?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/webSPELL/webSPELL/pull/209?approve=1'></a>&nbsp;<a href='https://github.com/webSPELL/webSPELL/pull/209'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by webSPELL-org** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/webSPELL/webSPELL/pull/209'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by Phhere** [comment]
We need a better naming schema for the called functions

**Comment by Pascalmh** [comment]
`_mysql.php` has to be reverted

File: install/index.php [hunk 1]
```diff
@@ -60,38 +60,44 @@
 
 ?>
 
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
```
Comment (Pascalmh) [inline install/index.php]: This is just `<!DOCTYPE html>` in HTML5

File: install/install.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            } else {
+                installSuccessfulCallback();
+            }
+        }
+    }).fail(function(result) {
+        installFailedCallback()
+    });
+}
+function installFailedCallback() {
+    $("#details_text").append("<font color='red'>Install failed</font>");
```
Comment (Pascalmh) [inline install/install.js]: The font-Tag is deprecated

http://www.w3.org/TR/html5/obsolete.html#non-conforming-features

put that stuff into the html, add classes and then use `.show()` instead. Why don't you use bootstrap here? http://getbootstrap.com/components/#alerts

File: install/index.php (hunk 1 above)
Comment (Phhere) [inline install/index.php]: As we have not changed much in the installer it should be like it is now

**Pull request merged**